src/mito2/src/memtable/bulk/part.rs

```diff
@@ -555,29 +593,113 @@ pub struct BulkPartMeta {
     pub region_metadata: RegionMetadataRef,
 }
 
+/// Metrics for encoding a part.
+#[derive(Default, Debug)]
+pub struct BulkPartEncodeMetrics {
+    /// Cost of iterating over the data.
+    pub iter_cost: Duration,
+    /// Cost of writing the data.
+    pub write_cost: Duration,
+    /// Size of data before encoding.
+    pub raw_size: usize,
+    /// Size of data after encoding.
+    pub encoded_size: usize,
+    /// Number of rows in part.
+    pub num_rows: usize,
+}
+
 pub struct BulkPartEncoder {
     metadata: RegionMetadataRef,
     row_group_size: usize,
     writer_props: Option<WriterProperties>,
 }
 
 impl BulkPartEncoder {
-    pub(crate) fn new(metadata: RegionMetadataRef, row_group_size: usize) -> BulkPartEncoder {
+    pub(crate) fn new(
+        metadata: RegionMetadataRef,
+        row_group_size: usize,
+    ) -> Result<BulkPartEncoder> {
+        // TODO(yingwen): Skip arrow schema if needed.
+        let json = metadata.to_json().context(InvalidMetadataSnafu)?;
+        let key_value_meta =
+            parquet::file::metadata::KeyValue::new(PARQUET_METADATA_KEY.to_string(), json);
+
+        // TODO(yingwen): Do we need compression?
```

Comment:
I think compression would be worth it, the size difference can be 10x

Reply:
We can measure this later. We have the raw size and encoded size in metrics. Compression might enlarge the merge cost.